### ISSUE TITLE
Test failed: JIT/opt/ObjectStackAllocation/ObjectStackAllocationTests/ObjectStackAllocationTests.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#27331]
**Job:**
[coreclr-jitstress #20191019.1](https://dev.azure.com/dnceng/public/_build/results?buildId=394826)

**Detail:**
linux x64:
https://helix.dot.net/api/2019-06-17/jobs/bf1f0311-43d8-43c3-a6f9-51269ad69f80/workitems/JIT/console
OSX x64:
https://helix.dot.net/api/2019-06-17/jobs/8159f61d-9813-48f6-b447-f50cdd09a276/workitems/JIT/console

**OS & Arch:**
linux x64
osx x64

**Mode:**
linux x64:
COMPlus_TieredCompilation=0
COMPlus_DbgEnableMiniDump=1
COMPlus_DbgMiniDumpName=$HELIX_DUMP_FOLDER/coredump.%d.dmp
COMPlus_JitStress=2

osx x64:
COMPlus_TieredCompilation=0
COMPlus_DbgEnableMiniDump=1
COMPlus_DbgMiniDumpName=$HELIX_DUMP_FOLDER/coredump.%d.dmp
COMPlus_JitStress=1

**Note:**
This failure **repros** on our osx x64 machine.
There is no 'Tests_Linux_x64_checked_jitstress' in artfacts.

**Log:**
```
JIT/opt/ObjectStackAllocation/ObjectStackAllocationTests/ObjectStackAllocationTests.sh [FAIL]
      
      Return code:      1
      Raw output file:      /home/helixbot/work/bf1f0311-43d8-43c3-a6f9-51269ad69f80/Work/7509fbd3-e1bb-4605-8f58-fc3ab6c77874/Exec/JIT/opt/Reports/JIT.opt/ObjectStackAllocation/ObjectStackAllocationTests/ObjectStackAllocationTests.output.txt
      Raw output:
      BEGIN EXECUTION
      /home/helixbot/work/bf1f0311-43d8-43c3-a6f9-51269ad69f80/Payload/corerun ObjectStackAllocationTests.dll ''
      SUCCESS (AllocateSimpleClassAndAddFields)
      FAILURE (AllocateSimpleClassesAndEQCompareThem): unexpected allocation of 48 bytes
      SUCCESS (AllocateSimpleClassesAndNECompareThem)
      SUCCESS (AllocateSimpleClassAndGetField)
      SUCCESS (AllocateClassWithNestedStructAndGetField)
      SUCCESS (AllocateClassWithNestedStructAndAddFields)
      SUCCESS (AllocateSimpleClassWithGCFieldAndAddFields)
      SUCCESS (AllocateSimpleClassAndAssignRefToAField)
      SUCCESS (TestMixOfReportingAndWriteBarriers)
      SUCCESS (AllocateSimpleClassAndCheckTypeNoHelper)
      SUCCESS (AllocateSimpleClassAndCheckTypeHelper)
      SUCCESS (AllocateSimpleClassAndCast)
      SUCCESS (BoxSimpleStructAndAddFields)
      Expected: 100
      Actual: 255
      END EXECUTION - FAILED
      Test Harness Exitcode is : 1
```

**Comment by VincentBu** [comment]
This test also failed in **windows x64/x86** in [coreclr-jitstress #20191020.1](https://dev.azure.com/dnceng/public/_build/results?buildId=395184)

**Mode:**
windows x64:
COMPlus_TieredCompilation=0
COMPlus_JitStress=2

windows x86:
COMPlus_TieredCompilation=1
COMPlus_JitStress=1

**Note:**
This failure **repros** on our windows x64 machine.

**Log:**
```
JIT\opt\ObjectStackAllocation\ObjectStackAllocationTests\ObjectStackAllocationTests.cmd [FAIL]
      
      Return code:      1
      Raw output file:      C:\dotnetbuild\work\2b4a8c80-1ce2-4327-978f-6b4be9ace338\Work\dfec452f-1d7a-468c-9dcc-f2ae83a465cc\Exec\JIT\opt\Reports\JIT.opt\ObjectStackAllocation\ObjectStackAllocationTests\ObjectStackAllocationTests.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\dotnetbuild\work\2b4a8c80-1ce2-4327-978f-6b4be9ace338\Payload\corerun.exe" ObjectStackAllocationTests.dll 
      SUCCESS (AllocateSimpleClassAndAddFields)
      FAILURE (AllocateSimpleClassesAndEQCompareThem): unexpected allocation of 32 bytes
      SUCCESS (AllocateSimpleClassesAndNECompareThem)
      SUCCESS (AllocateSimpleClassAndGetField)
      SUCCESS (AllocateClassWithNestedStructAndGetField)
      SUCCESS (AllocateClassWithNestedStructAndAddFields)
      SUCCESS (AllocateSimpleClassWithGCFieldAndAddFields)
      SUCCESS (AllocateSimpleClassAndAssignRefToAField)
      SUCCESS (TestMixOfReportingAndWriteBarriers)
      SUCCESS (AllocateSimpleClassAndCheckTypeNoHelper)
      SUCCESS (AllocateSimpleClassAndCheckTypeHelper)
      SUCCESS (AllocateSimpleClassAndCast)
      SUCCESS (BoxSimpleStructAndAddFields)
      Expected: 100
      Actual: -1
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
```

**Comment by erozenfeld** [comment]
Fixed by #27344 .
